src/NuGet.Core/NuGet.ProjectModel/Utf8JsonReaderExtensions.cs

```diff
@@ -0,0 +1,231 @@
+        //    if (reader.TokenType == JsonTokenType.StartArray)
+        //    {
+        //        while (ReadNextToken(ref reader) && reader.TokenType != JsonTokenType.EndArray)
+        //        {
+        //            string value = ReadTokenAsString(ref reader);
+
+        //            strings = strings ?? new List<string>();
+
+        //            strings.Add(value);
+        //        }
+        //    }
+
+        //    return strings;
+        //}
+
+        //internal static void ReadArrayOfObjects<T1, T2>(this ref Utf8JsonReader reader, JsonSerializerOptions options, IList<T2> objectList) where T1 : T2
+        //{
+        //    if (objectList is null)
+        //    {
+        //        return;
+        //    }
+
+        //    var type = typeof(T1);
+        //    var objectConverter = (JsonConverter<T1>)options.GetConverter(type);
+
+        //    if (ReadNextToken(ref reader) && reader.TokenType == JsonTokenType.StartArray)
+        //    {
+        //        while (ReadNextToken(ref reader) && reader.TokenType != JsonTokenType.EndArray)
+        //        {
+        //            var convertedObject = objectConverter.Read(ref reader, type, options);
+        //            if (convertedObject != null)
+        //            {
+        //                objectList.Add(convertedObject);
+        //            }
+        //        }
+        //    }
+
+        //}
+
+        //internal static IReadOnlyList<string> ReadNextStringOrArrayOfStringsAsReadOnlyList(this ref Utf8JsonReader reader)
+        //{
+        //    if (ReadNextToken(ref reader))
+        //    {
+        //        switch (reader.TokenType)
+        //        {
+        //            case JsonTokenType.String:
+        //                return new[] { (string)reader.GetString() };
+
+        //            case JsonTokenType.StartArray:
+        //                return ReadStringArrayAsReadOnlyListFromArrayStart(ref reader);
+
+        //            case JsonTokenType.StartObject:
+        //                return null;
+        //        }
+        //    }
+
+        //    return null;
+        //}
+
+        //internal static IReadOnlyList<string> ReadStringArrayAsReadOnlyListFromArrayStart(this ref Utf8JsonReader reader)
+        //{
+        //    List<string> strings = null;
+
+        //    while (ReadNextToken(ref reader) && reader.TokenType != JsonTokenType.EndArray)
+        //    {
+        //        string value = ReadTokenAsString(ref reader);
+
+        //        strings = strings ?? new List<string>();
+
+        //        strings.Add(value);
+        //    }
+
+        //    return (IReadOnlyList<string>)strings ?? Array.Empty<string>();
+        //}
+
+        internal static string ReadTokenAsString(this ref Utf8JsonReader reader)
+        {
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.True:
+                case JsonTokenType.False:
+                    return reader.GetBoolean().ToString();
```

Comment:
This could probably return `bool.TrueString` or `bool.FalseString` instead to avoid allocations